src/rust/engine/process_execution/src/immutable_inputs.rs

```diff
@@ -42,20 +46,41 @@ impl ImmutableInputs {
     let value: Result<_, String> = cell
       .get_or_try_init(async {
         let digest_str = digest.hash.to_hex();
-
         let path = self.workdir.path().join(digest_str);
         if let Ok(meta) = tokio::fs::metadata(&path).await {
-          // TODO: If this error triggers, it indicates that we have previously checked out this
-          // directory, either due to a race condition, or due to a previous failure to
-          // materialize. See https://github.com/pantsbuild/pants/issues/13899
+          // If this error triggers, it indicates that we have previously checked out this
+          // directory, and OnceCell double checked locking is broken.
+          // We can remove this sanity-check if we gain confidence that OnceCell does its job.
+          // See: https://github.com/pantsbuild/pants/issues/13899
           return Err(format!(
```

Comment:
N.B.: This Err did not trip but the rename below did:
```
thread 'local_tests::immutable_inputs' panicked at 'called `Result::unwrap()` on an `Err` value: "Failed to move materialized chroot for Digest { hash: Fingerprint<a80c52a2b4be9145efe3a05bbdd619e959cf78abd3b580977f391fcb1d7c54e9>, size_bytes: 84 } from \"/Users/runner/work/_temp/.tmpLeJx5x/immutable_inputs7VXF48/.tmpgKgLW1\" to \"/Users/runner/work/_temp/.tmpLeJx5x/immutable_inputs7VXF48/a80c52a2b4be9145efe3a05bbdd619e959cf78abd3b580977f391fcb1d7c54e9\": Permission denied (os error 13)"', process_execution/src/local_tests.rs:634:4
```

Which makes sense. If we're racy, we're racy on any given line.